src/promql/src/functions/holt_winters.rs

```diff
@@ -142,6 +119,36 @@ impl HoltWinters {
 
         // calculation
         let mut result_array = Vec::with_capacity(ts_range.len());
+
+        let get_sf = |i| {
```

Comment:
Non-blocking: Extract a macro or function to avoid code duplication during conversation handling.